app/src/main/java/com/github/se/eventradar/model/event/AllEventOverview.kt

```diff
@@ -0,0 +1,45 @@
+package com.github.se.eventradar.model.event
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.google.firebase.Firebase
+import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.firestore.firestore
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.ZoneId
+
+class AllEventOverview(private val db: FirebaseFirestore = Firebase.firestore): ViewModel() {
```

Comment:
This is not a good name. What is "all events"? `ViewModel` should be in the name.